src/plugins/WebOfTrust/util/DelayedBackgroundJobFactory.java

```diff
@@ -0,0 +1,79 @@
+package plugins.WebOfTrust.util;
+
+import java.util.WeakHashMap;
+
+import freenet.support.Executor;
+import freenet.support.Ticker;
+
+/**
+ * Factory for {@link DelayedBackgroundJob}s which allows all background jobs constructed using
+ * an instance of this class to be terminated with a single method invocation.
+ *
+ * @author bertm
+ * @see DelayedBackgroundJob
+ */
+public class DelayedBackgroundJobFactory {
+    private static final Object PLACEHOLDER = new Object();
+
+    /** Set of all live (i.e. not garbage collected) background jobs created by this instance. */
+    private final WeakHashMap<BackgroundJob, Object> aliveJobSet;
```

Comment:
ACK to keep as is. Should have prefixed this with "Just FYI, no change required", sorry :)